WordPress/gutenberg
# [Data Views][Grid Layout] Adding semantics and basic keyboard navigation

See also: #59188

## What?

Adds structural semantics to grid layout, and improves keyboard navigation.


## Why?

Grid layouts are currently semantically ambiguous, with no structural scaffolding to denote which fields relate to each other.


## How?

- Grid container is given `role="grid"`
- Grid items are given `role="gridcell"`
- Grid items are appropriately wrapped in `role="row"` elements
- Grid items can be navigated using arrow keys


## Testing Instructions

- Visually there should be little difference
- For mouse users there should be no difference
- The grid items should wrap when there is insufficent horizontal space


### Testing Instructions for Keyboard
- Tabbing to the grid should select the first grid item
- Arrow keys should navigate around the grid
- Additional tabbing should focus on the child components
- <kbd>Escape</kbd> should focus the current grid item
